src/polynomial.h

```diff
@@ -60,7 +60,7 @@ class Polynomial {
     void copy(Polynomial<T>* src);
     void copy(Polynomial<T>* src, T offset);
     int degree();
-    Term<T> get_terms();
+    Term<T>* get_terms();
```

Comment:
Do you need a write access to `terms`? Otherwise you could return a const reference (and declare the method as **const**)